collector/info_schema_processlist.go

```diff
@@ -189,83 +109,102 @@ func (ScrapeProcesslist) Scrape(ctx context.Context, db *sql.DB, ch chan<- prome
 	defer processlistRows.Close()
 
 	var (
-		user      string
-		host      string
-		command   string
-		state     string
-		processes uint32
-		time      uint32
+		user    string
+		host    string
+		command string
+		state   string
+		count   uint32
+		time    uint32
 	)
-	stateCounts := make(map[string]uint32, len(threadStateCounterMap))
-	stateTime := make(map[string]uint32, len(threadStateCounterMap))
-	hostCount := make(map[string]uint32)
-	userCount := make(map[string]uint32)
-	for k, v := range threadStateCounterMap {
-		stateCounts[k] = v
-		stateTime[k] = v
-	}
+	// Define maps
+	stateCounts := make(map[string]map[string]uint32)
+	stateTime := make(map[string]map[string]uint32)
+	stateHostCounts := make(map[string]uint32)
+	stateUserCounts := make(map[string]uint32)
 
 	for processlistRows.Next() {
-		err = processlistRows.Scan(&user, &host, &command, &state, &processes, &time)
+		err = processlistRows.Scan(&user, &host, &command, &state, &count, &time)
 		if err != nil {
 			return err
 		}
-		realState := deriveThreadState(command, state)
-		stateCounts[realState] += processes
-		stateTime[realState] += time
-		hostCount[host] = hostCount[host] + processes
-		userCount[user] = userCount[user] + processes
-	}
+		command = sanitizeState(command)
+		state = sanitizeState(state)
+		if host == "" {
+			host = "blank"
```

Comment:
I think we would prefer to keep an empty string here, or `"unknown"` rather than `"blank"`.